tests/foreman/cli/test_webhook.py

```diff
@@ -0,0 +1,216 @@
+
+        return Box(Webhook.create(options))
+
+    return partial(_create_webhook, org=class_org, loc=class_location)
+
+
+def assert_created(options, hook):
+    for option in options.items():
+        if not option[0] in ['event', 'organization-id', 'location-id']:
+            assert hook[option[0]] == option[1]
+
+
+class TestWebhook:
+    @pytest.mark.tier2
+    def test_negative_invalid_event(self, webhook_factory):
+        """Test negative webhook creation with an invalid event
+
+        :id: 60cd456a-9943-45cb-a72e-23a83a691499
+
+        :expectedresults: Webhook is not created
+
+        :CaseImportance: High
+        """
+        with pytest.raises(CLIError):
+            webhook_factory(options={'event': 'invalid_event'})
+
+    @pytest.mark.tier2
+    @pytest.mark.parametrize('event', **parametrized(WEBHOOK_EVENTS))
+    def test_positive_valid_event(self, event, webhook_factory):
+        """Test positive webhook creation with a valid event
+
+        :id: 9b505f1b-7ee1-4362-b44c-f3107d043a05
+
+        :expectedresults: Webhook is created with event specified
+
+        :CaseImportance: High
+        """
+        hook = webhook_factory(options={'event': event})
+        assert event in hook.event
+
+    @pytest.mark.tier2
+    def test_negative_invalid_method(self, webhook_factory):
+        """Test negative webhook creation with an invalid HTTP method
+
+        :id: 573be312-7bf3-4d9e-aca1-e5cac810d04b
+
+        :expectedresults: Webhook is not created
+
+        :CaseImportance: High
+        """
+        with pytest.raises(CLIError):
+            webhook_factory(options={'http-method': 'NONE'})
+
+    @pytest.mark.tier2
+    @pytest.mark.parametrize('method', **parametrized(WEBHOOK_METHODS))
+    def test_positive_valid_method(self, method, webhook_factory):
+        """Test positive webhook creation with a valid HTTP method
+
+        :id: cf8f276a-d21e-44d0-92f2-657232240c7e
+
+        :expectedresults: Webhook is created with specified method
+
+        :CaseImportance: High
+        """
+        hook = webhook_factory(options={'http-method': method})
+        assert hook['http-method'] == method
+
+    @pytest.mark.tier1
+    def test_positive_update_hook(self, webhook_factory):
+        original_options = {
+            'name': gen_alphanumeric(),
+            'http-method': 'GET',
+            'event': WEBHOOK_EVENTS[0],
+            'target-url': 'http://localhost/original',
+        }
+        hook = webhook_factory(options=original_options)
+        assert_created(original_options, hook)
+        update_options = {
+            'name': hook.name,
+            'http-method': 'POST',
+            'event': WEBHOOK_EVENTS[0],
+            'target-url': 'http://localhost/original',
+        }
+        updated_hook = Webhook.update(update_options)
+        assert f'Webhook [{hook.name}] updated.' in updated_hook[0]['message']
+
+    @pytest.mark.tier1
+    def test_positive_read_hook(self, webhook_factory):
```

Comment:
Isn't this covered in `test_positive_end_to_end`?, do we need a separate test for it?